market-generator/src/main/java/com/market/banica/generator/service/MarketSubscriptionImpl.java

```diff
@@ -0,0 +1,97 @@
+package com.market.banica.generator.service;
+
+import com.market.MarketDataRequest;
+import com.market.TickResponse;
+import com.market.banica.generator.exception.NoSuchGoodException;
+import io.grpc.StatusRuntimeException;
+import io.grpc.netty.shaded.io.netty.util.internal.StringUtil;
+import io.grpc.stub.StreamObserver;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+@Component
+public class MarketSubscriptionImpl implements Subscription<MarketDataRequest, TickResponse> {
```

Comment:
Rename to SubscriptionManager or MarketSubscriptionManager depending on the interface name.